### PR TITLE
Fix #18426 error when url route is empty in `yii\widgets\Menu`

### LINKED ISSUE #18426: PHP Notice 'Uninitialized string offset: 0' in `yii\widgets\Menu`
Get notice`Uninitialized string offset 0 in /vendor/yiisoft/yii2/widgets/Menu.php:309` when using `''` as a url. Since PHP 8 has `E_ALL` turned on by default, it will be a problem.

### What steps will reproduce the problem?
```php
echo Menu::widget([
    'items' => [
        ['label' => 'Some Label', 'url' => ['', '#' => 'someid']]
    ]
]);
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.39.3
| PHP version      | 7.4, 8.0
| Operating system | Linux, MacOS

### PR DESCRIPTION
**Pull request by stevekr**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #18426

### CODE REVIEW

**Review by bizley — APPROVED** [review]

**Comment by bizley** [comment]
Please add line in changelog as well, thanks!

**Comment by stevekr** [comment]
> Please add line in changelog as well, thanks!

Done

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Yii Framework 2 Change Log
 - Enh #18394: Add support for setting `yii\web\Response::$stream` to a callable (brandonkelly)
 - Bug #18406: Fix PDO exception when committing or rolling back an autocommitted transaction in PHP 8 (brandonkelly)
 - Bug #18287: Fix the OUTPUT got SQL syntax error if the column name is MSSQL keyword e.g key (darkdef)
+- Bug #18426: Fixed error when url route is empty in `yii\widgets\Menu` (stevekr)
```
Comment (bizley) [inline framework/CHANGELOG.md:15]:
```suggestion
- Bug #18426: Fix check for route's leading slash in `yii\widgets\Menu` (stevekr)
```

**Review by bizley — CHANGES_REQUESTED** [review]

**Review by bizley — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**